### PR TITLE
Define a BlockRenderer object that Geras, Thrasos, and Zelos extend

### LINKED ISSUE #2966: Define a BlockRenderer object that Geras, Thrasos, and Zelos extend
The object should define init and render functions and set up the correct constant provider, drawer, etc.

### PR DESCRIPTION
**Pull request by rachel-fenichel**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

Fixes #2966

### Proposed Changes
Defines a base renderer that specifies its constantProvider, renderInfo, drawer, and debugger classes.  It has `init` and `render` functions.  The child classes mostly just change the constructor; geras also adds a highlight provider in init.

### Reason for Changes

Gets us closer to removing block_rendering.js entirely
### Test Coverage
Tried all three renderers in the playground.

### Additional Information
`Blockly.blockRendering.renderer.render(block);` is a very silly line of code.

### CODE REVIEW

File: core/renderers/geras/renderer.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ */
+'use strict';
+
+goog.provide('Blockly.geras.Renderer');
+
+goog.require('Blockly.blockRendering.ConstantProvider');
+goog.require('Blockly.blockRendering.Debug');
+goog.require('Blockly.blockRendering.Renderer');
+goog.require('Blockly.geras.Drawer');
+goog.require('Blockly.geras.RenderInfo');
```
Comment (samelhusseini) [inline core/renderers/geras/renderer.js]: require Blockly.geras.HighlightConstantProvider

File: core/renderers/common/renderer.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ * @author fenichel@google.com (Rachel Fenichel)
+ */
+'use strict';
+
+goog.provide('Blockly.blockRendering.Renderer');
+
+goog.require('Blockly.blockRendering.ConstantProvider');
+goog.require('Blockly.blockRendering.Debug');
+goog.require('Blockly.blockRendering.Drawer');
+goog.require('Blockly.blockRendering.RenderInfo');
```
Comment (samelhusseini) [inline core/renderers/common/renderer.js]: require Blockly.blockRendering.Debugger

Comment (samelhusseini) [reply core/renderers/common/renderer.js]: actually, that's not a thing. typo here Blockly.blockRendering.Debugger should be Blockly.blockRendering.Debug

File: core/renderers/common/renderer.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+/**
+ * The base class for a block renderer.
+ * @package
+ * @constructor
+ */
+Blockly.blockRendering.Renderer = function() {
+  this.constantProvider = Blockly.blockRendering.ConstantProvider;
+  this.renderInfo = Blockly.blockRendering.RenderInfo;
+  this.drawer = Blockly.blockRendering.Drawer;
+  this.debugger = Blockly.blockRendering.Debugger;
```
Comment (samelhusseini) [inline core/renderers/common/renderer.js]: should be Blockly.blockRendering.Debug

File: core/renderers/geras/renderer.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+/**
+ * Initialize the renderer.  Geras has a highlight provider in addition to
+ * the normal constant provider.
+ * @package
+ */
+Blockly.geras.Renderer.prototype.init = function() {
+  Blockly.geras.Renderer.superClass_.init.call(this);
+  this.highlightConstants =
+      new Blockly.geras.HighlightConstantProvider(this.constants);
```
Comment (samelhusseini) [inline core/renderers/geras/renderer.js]: should this.constants be passed in? Blockly.geras.HighlightConstantProvider isn't expecting a parameter.

**Comment by rachel-fenichel** [comment]
K, I think I got all of them.

**Pull request merged**